GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -39,8 +39,13 @@ public enum RevisionGraphDrawStyleEnum
         HighlightSelected
     }
 
+    public interface IUserScriptMenuBuilder
+    {
+        void Build(ContextMenuStrip contextMenu);
+    }
+
     [DefaultEvent("DoubleClick")]
-    public sealed partial class RevisionGridControl : GitModuleControl
+    public sealed partial class RevisionGridControl : GitModuleControl, IUserScriptMenuBuilder
```

Comment:
I don't particularly mind the implementation however I don't like the fact we are adding more functionality to this class, it is already too big.
Let's have a dedicated `public sealed class UserScriptMenuBuilder : IUserScriptMenuBuilder`, that `RevisionGridControl` will depend on (via private field).

This way we will decouple the building of the script menu from the control, and other controls could leverage the same (e.g. #6135 is a perfect candidate for the menu).

The implementation will also be significantly more testable.